utilities/transactions/write_prepared_txn_db.cc

```diff
@@ -157,7 +157,9 @@ Status WritePreparedTxnDB::WriteInternal(const WriteOptions& write_options_orig,
     // TODO(myabandeh): add an option to allow user skipping this cost
     SubBatchCounter counter(*GetCFComparatorMap());
     auto s = batch->Iterate(&counter);
-    assert(s.ok());
+    if (!s.ok()) {
```

Comment:
The relevant case this is non-OK is when `batch` contains a range deletion. That case is possible through API misuse (the tests exercise this path) so I think it should be a runtime check instead of an assertion.